File: leetcode1807.cpp
#include "bits/stdc++.h"
using namespace std;
class Solution {
public:
    string evaluate(string s, vector<vector<string>>& knowledge) {
        string a;
        map<string,string> mapp;
        std::for_each(knowledge.begin(), knowledge.end(), [&](const auto &item) {
            mapp[item[0]] = item[1];
        });
        for (int i = 0;i < s.size();i ++) {
            if (s[i] != '(')
                a.push_back(s[i]);
            else {
                string b;
                i ++;
                while (s[i] != ')') {
                    b.push_back(s[i ++]);
                }
                if (!mapp[b].empty())
                    a.append(mapp[b]);
                else a.push_back('?');
            }
        }
        return a;
    }
};
int main() {
    string s = "(name)is(age)yearsold";
    vector<vector<string>> val = {{"name","bob"},{"age","two"}};
    cout<<(new Solution)->evaluate(s,val);
    return 0;
}

File: leetcode1807.go
package main

import (
	"strings"
)

func evaluate(s string, knowledge [][]string) string {
	a := strings.Builder{}
	mapp := make(map[string]string)
	for _, i := range knowledge {
		mapp[i[0]] = i[1]
	}
	for i := 0; i < len(s); i++ {
		if s[i] != '(' {
			a.WriteByte(s[i])
		} else {
			b := strings.Builder{}
			i++
			for s[i] != ')' {
				b.WriteByte(s[i])
				i++
			}
			s2, err := mapp[b.String()]
			if err != false {
				a.WriteString(s2)
			} else {
				a.WriteByte('?')
			}
		}
	}
	return a.String()
}
func main() {
	s := "(name)is(age)yearsold"
	b := [][]string{{"name", "bob"}, {"age", "two"}}
	println(evaluate(s, b))
}
